src/core/useCases/locale/locale.spec.ts

```diff
@@ -0,0 +1,91 @@
+import { configureStore } from '../../configureStore'
+import { PatialAppDependencies } from '../Dependencies'
+import { PartialAppState, defaultInitialAppState, reducers } from '../reducers'
+import { TestDeviceLocale } from './TestDeviceLocale'
+import { TestLocaleStorage } from './TestLocaleStorage'
+import { fakeLocale1, fakeLocale2 } from './__mocks__'
+import { publicActions } from './locale.actions'
+import { localeSaga } from './locale.saga'
+import {
+  selectLocale,
+} from './locale.selectors'
+
+function configureStoreWithLocale(
+  params: {
+    dependencies?: PatialAppDependencies;
+    initialAppState?: PartialAppState;
+  },
+) {
+  const { initialAppState, dependencies } = params
+
+  return configureStore({
+    reducers,
+    initialState: {
+      ...defaultInitialAppState,
+      ...initialAppState,
+    },
+    dependencies,
+    sagas: [localeSaga],
+  })
+}
+
+describe('Locale', () => {
+  it(`
+    Given the site has loaded
+    If user has no local saved in storage
+    Then the default locale should be the browser's locale
+      And the locale should be saved in storage
```

Comment:
On doit avoir un seul Then / And par `expect` . Ici tu as un seul expect, donc un seul `Then`

```suggestion
    Then the selected locale should be the device locale
```

Par contre, tu devrais ajouter un test pour vérifier que `localeStorage.storeLocale` a bien été appelé avec la Locale en question. Tu peux regarder ici pour l'exemple avec auth user: https://github.com/ReseauEntourage/entourage-webapp/blob/develop/src/core/useCases/authUser/authUser.spec.ts#L750